wikidata/client.py

```diff
@@ -179,19 +180,28 @@ def decode_datavalue(self,
 
     def request(self, path: str) -> Union[
         bool, int, float, str,
-        Mapping[
-            str,
-            Union[bool, int, float, str, Mapping[str, object], Sequence]
-        ],
-        Sequence[Union[bool, int, float, str, Mapping[str, object], Sequence]]
+        Mapping[str, Union[bool, int, float, str,
+                           Mapping[str, object], Sequence]],
+        Sequence[Union[bool, int, float, str, Mapping[str, object], Sequence]],
+        None
     ]:
         logger = logging.getLogger(__name__ + '.Client.request')
         url = urllib.parse.urljoin(self.base_url, path)
         result = self.cache_policy.get(CacheKey(url))
         if result is None:
             logger.debug('%r: no cache; make a request...', url)
-            response = self.opener.open(url)
-            buffer_ = io.TextIOWrapper(response, encoding='utf-8')
+            try:
+                response = self.opener.open(url)
+            except urllib.error.HTTPError as e:
+                logger.debug('HTTP error code: %s', e.code, exc_info=True)
+                response = e.fp.read().decode('utf-8')  # type: ignore
+                if 'Invalid ID' in response:
+                    return 'Non-Existent'
```

Comment:
You added `None` to the arguments of `Union[...]` which is its return type.  So didn't you intend to return `None` for non-existent entities?